org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/PreferenceManager.java

```diff
@@ -259,6 +260,10 @@ public void update(Preferences preferences) {
 		if (!oldPreferences.getFilesAssociations().equals(preferences.getFilesAssociations())) {
 			configureContentTypes(preferences);
 		}
+		
+		// update call hierachy test code filer
+		final boolean filterTestCode = this.preferences.getSearchScope() == SearchScope.main;
+		eclipsePreferences.put("PREF_FILTER_TESTCODE", String.valueOf(filterTestCode));
```

Comment:
Right. Interesting approach. Looks like it came from the days of splitting up core/ui https://github.com/eclipse-jdt/eclipse.jdt.ui/blob/65be1d3f04da8f3ff22701b3bf6a3bcaba7a52c8/org.eclipse.jdt.core.manipulation/core%20extension/org/eclipse/jdt/internal/corext/callhierarchy/CallHierarchyCore.java#L72-L74 but ok.